app/updater.js

```diff
@@ -212,8 +214,21 @@ var downloadHandler = (err, metadata) => {
 }
 
 // Make network request to check for an available update
-exports.checkForUpdate = (verbose) => {
-  const updateStatus = AppStore.getState().getIn(['updates', 'status'])
+exports.checkForUpdate = (verbose, skipReferral = false) => {
+  const state = AppStore.getState()
+
+  // check for referral 30 days
+  if (!skipReferral && !ledgerState.getInfoProp(state, 'referralTimestamp')) {
+    const installTime = state.get('firstRunTimestamp')
+    const month = ledgerUtil.milliseconds.day * 30
+
+    if (installTime + month < new Date().getTime()) {
```

Comment:
this function will check for referral activity every 30 days. i thought it was only supposed to check referral activity once after 30 days and then never check again.

Reply:
server can be in a different zone or not aligned with the browsers, so we need to check until referral is successful. We have `referralTimestamp` above which will not trigger this checks anymore as soon as we get green light from the server. We can add amount of attempts after which we will just stop checking?

Reply:
Where is `referralTimestamp` set to a false value after getting a green light from the server?

Reply:
we don't set it to false, we actually set value (timestamp) to it in the ledger reducer